react/components/EXPERIMENTAL_Table/hooks/useHiddenColumns.ts

```diff
@@ -0,0 +1,41 @@
+import { useState, useMemo } from 'react'
+
+const getHiddenColumns = (columns: Array<Column>): Array<string> => {
+  return columns.filter(col => col.hidden).map(col => col.id)
+}
+
+const useHiddenColumns = (columns: Array<Column>) => {
+  const [hiddenColumns, setHiddenColumns] = useState(getHiddenColumns(columns))
+
+  const visibleColumns = useMemo(() => {
+    const reducer = (acc: Array<string>, col: Column) =>
+      hiddenColumns.includes(col.id) ? acc : [...acc, col]
+
+    return columns.reduce(reducer, [])
+  }, [hiddenColumns, columns])
+
+  const toggleColumn = (id: string) => {
```

Comment:
I think it's good practice to *always* memoize callbacks and values from custom hooks, so the component that uses them can safely add them to the dependencies of an `useEffect`

Reply:
✅